go/vt/vtgate/vstream_manager.go

```diff
@@ -411,18 +411,46 @@ func (vs *vstream) alignStreams(ctx context.Context, event *binlogdatapb.VEvent,
 	}
 }
 
-func (vs *vstream) getCells() []string {
+// getCells determines the availability zones to select tablets from
+// 2 scenarios:
+//
+// 1. No cells specified by the client via the gRPC request:
+// Tablets from the local cell AND the cell alias that the VTGate's local cell belongs to will be selected.
+// The local cell of the VTGate will take precendence over any other cell in the alias.
+//
+// 2. Cells are specified by the client via the gRPC request
+// These cells will take precendence over the default cell and its alias and only tablets belonging to
+// the specified cells will be selected.
+// If the "local" tag is passed in as an option in the list of optCells,
+// the local cell of the VTGate will take precedence over any other cell specified.
+func (vs *vstream) getCells(ctx context.Context) []string {
 	var cells []string
 	if vs.optCells != "" {
-		for _, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+		for i, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+			// if the local tag is passed in, we must give local cell priority
+			// during tablet selection. Append the VTGate's local cell to the list of cells
+			if i == 0 && cell == "local" {
+				cells = append(cells, fmt.Sprintf("local:%s", vs.vsm.cell))
+				continue
+			}
 			cells = append(cells, strings.TrimSpace(cell))
 		}
 	}
 
+	// if no cell override provided in gRPC request, perform cell alias fallback
 	if len(cells) == 0 {
-		// use the vtgate's cell by default
-		cells = append(cells, vs.vsm.cell)
+		log.Info("[VSTREAM MANAGER] no cells specified by client, falling back to alias...")
+		// append the alias this cell belongs to, otherwise appends the vtgate's cell
+		alias := topo.GetAliasByCell(ctx, vs.ts, vs.vsm.cell)
+		// an alias was actually found
+		if alias != vs.vsm.cell {
+			// send in the vtgate's cell for local cell preference
+			cells = append(cells, fmt.Sprintf("local:%s", vs.vsm.cell))
```

Comment:
it's unclear why we need to lookup the cell alias if the tablet picker already does cell alias lookup here: https://github.com/slackhq/vitess/blob/5563eb126e91457ef0dde6e51f6375efd0532a01/go/vt/discovery/tablet_picker.go#L195-L200

Reply:
It only does a cell lookup if the string passed into "tp.cells" is not a valid cell: https://github.com/slackhq/vitess/blob/5563eb126e91457ef0dde6e51f6375efd0532a01/go/vt/discovery/tablet_picker.go#L191